fontforge/mem.c

```diff
@@ -79,17 +79,22 @@ int get3byte(FILE *ttf) {
 }
 
 int32 getlong(FILE *ttf) {
-	int ch1 = getc(ttf);
-	int ch2 = getc(ttf);
-	int ch3 = getc(ttf);
-	int ch4 = getc(ttf);
+	unsigned int ch1 = getc(ttf);
```

Comment:
Hold up, these changes look suspicious to me. This for sure would raise a signed-unsigned conversion warning if we actually bothered to turn up the warning levels.

ubsan in a sense is likely correct anyway, as we're not checking for EOFs on all the bytes.

Reply:
So they should all be checked for EOF? That makes sense.

Reply:
I took a closer look at this and the EOF check at the end is fine, because if you don't get an EOF there, that implies that all the other reads succeeded (assuming a sane implementation of getc). 

However, both with and without these changes, there is still UB (or rather, implementation defined behaviour): https://stackoverflow.com/questions/13636397/unsigned-to-signed-conversion-in-c. It's just in this case, ubsan isn't complaining.

gcc does what the code (as written) expects it to, hence why nothing has fallen over.